map2loop/interpolators.py

```diff
@@ -0,0 +1,443 @@
+from abc import ABC, abstractmethod
+from typing import Any, Union
+import beartype
+import numpy
+from numpy import ndarray
+from scipy.interpolate import Rbf, LinearNDInterpolator
+from .utils import strike_dip_vector, setup_grid
+import pandas
+
+
+class Interpolator(ABC):
+    """
+    Base Class of Interpolator used to force structure of Interpolator
+
+    Args:
+        ABC (ABC): Derived from Abstract Base Class
+    """
+
+    def __init__(self):
+        """
+        Initialiser of for Interpolator
+        """
+        self.interpolator_label = "InterpolatorBaseClass"
+
+    def type(self):
+        """
+        Getter for subclass type label
+
+        Returns:
+            str: Name of subclass
+        """
+        return self.interpolator_label
+
+    @beartype.beartype
+    @abstractmethod
+    def setup_interpolation(self, structure_data: pandas.DataFrame):
+        """
+        abstract method to setup interpolation (abstract method)
+
+        Args:
+            structure_data (pandas.DataFrame): sampled structural data
+        """
+        pass
+
+    @beartype.beartype
+    @abstractmethod
+    def setup_grid(self, bounding_box: dict):
+        """
+        abstract method to setup an XY grid (abstract method)
+
+        Args:
+            bounding_box (dict): a dictionary containing the bounding box of the map data
```

Comment:
should there be a specification for the structure of the dict bbox? something like: 

{
'xmin': 0, 
'ymin': 0, 
'xmax': 1, 
'ymax':1
}

Reply:
docstring updated to include dictionary format enforced in mapdata.py